journal/events/consumer.go

```diff
@@ -11,16 +11,19 @@ import (
 	"github.com/absmach/supermq/journal"
 	"github.com/absmach/supermq/pkg/events"
 	"github.com/absmach/supermq/pkg/events/store"
+	"github.com/absmach/supermq/pkg/messaging"
 )
 
 var ErrMissingOccurredAt = errors.New("missing occurred_at")
 
 // Start method starts consuming messages received from Event store.
 func Start(ctx context.Context, consumer string, sub events.Subscriber, service journal.Service) error {
 	subCfg := events.SubscriberConfig{
-		Consumer: consumer,
-		Stream:   store.StreamAllEvents,
-		Handler:  Handle(service),
+		Consumer:       consumer,
```

Comment:
@felixgateru  
For Example, If the service replication increase, the newly started service will consume all the messages from first ? 
Could you please confirm this case with NATS Jetstream consumers

Reply:
According to https://docs.nats.io/nats-concepts/jetstream/streams the behaviour is based on the use of a durable name.
If the consumer durable name is the same among replicated consumers then they will share messages and each will not have the same messages but if the consumer durable name is unique the deliver all policy will apply for each consumer and they will each have the exact same messages which will include those from the beginning of the stream.

Reply:
Our events subscriber is using [LimitsPolicy](https://github.com/absmach/supermq/blob/17a884f7e99f53fa7ce3bf7705ab1279d8b7c445/pkg/events/nats/subscriber.go#L29), shouldn't we use `WorkQueuePolicy` instead?

Reply:
I don't think the policy would be suitable for us as we have multiple overlapping consumers i.e
![image](https://github.com/user-attachments/assets/3b502642-6d13-4fd1-9099-9071b06050e1)

Reply:
@dborovcanin  This LimitsPolicy for creating a stream, For Consumer we use this configuration https://github.com/absmach/supermq/blob/757ad31f921d7a05df8f1f4255cb41a52ce5f233/pkg/events/nats/subscriber.go#L67-L76 and this will call the NATS subscriber at here https://github.com/absmach/supermq/blob/602025b48bc008d8ec89be55e8ed228f1cebc4cb/pkg/messaging/nats/pubsub.go#L75-L114

Reply:
What happens if the consumer is deleted ? 
Does the messages will be start from the first ?

Reply:
If the Not Acked messages increase in consumer, Then we will delete consumer and create new consumer. 
To avoid Not Acked Messages again getting back to queue
So if we have All Messages policy , I think again all the messages from stream will get into consumer. Again the error message will get into queue



Another problem , If we delete the consumer and again the all the messages start flow into journal, 

Journal service might throw error, because these messages are already added in journal database. 

So we should handle this case in Journal service too.

Reply:
We can add this case only if journal service can able to handle the duplicate messages. 
@felixgateru  At present if journal could not able to handle duplicate message, then please revert these changes and we will have this in new PR along with journal service feature to handle duplicate message.